### PR TITLE
feat(groups): prevent removing the last admin via UI (#1257)

### LINKED ISSUE #1257: Cannot remove the last admin from a group (protection missing)
## Summary

The UI currently allows removing any member from a group, including the last remaining admin. If this goes through, the group becomes unmanageable — no one can invite new members, change roles, or delete the group.

## Expected behavior

- **Backend validation**: removing the last admin must return HTTP 400 / 409 with a clear error message.
- **Frontend**: the Remove button on the last admin should be disabled (with tooltip: "Cannot remove the last admin — promote another member first or delete the group").
- The group must always have **at least one admin**.

## Related

- See separate issue about leaving a group as the last admin (same class of problem, different entry point).
- See UI/UX meta-issue about consistent validation feedback.

## Acceptance criteria

- [ ] Backend refuses the operation with descriptive error
- [ ] Frontend disables the Remove action on the last admin
- [ ] Error surfaced clearly in UI (not silently swallowed)
- [ ] Covered by e2e tests: attempting to remove the last admin fails

### PR DESCRIPTION
**Pull request by denisvmedia**
Closes #1257.

## Summary

Parallel to #1259/#1270, which gated the "leave group" path, this closes the same gap for the admin-facing **Remove member** path. The backend already rejects the request with `422 ErrLastAdmin` (`GroupService.RemoveMember`); this PR prevents the doomed request from ever being submitted and explains *why* to mouse, keyboard, and screen-reader users.

## Backend

- `TestGroupService_RemoveMember_LastAdminProtection` upgraded from `qt.IsNotNil` to `qt.ErrorIs(err, services.ErrLastAdmin)` — the invariant must be enforced by the specific sentinel, not just "any error." Also extends coverage to:
  - Non-admin member removal when only one admin exists (invariant not at risk)
  - Admin removal unblocked once a second admin exists

  Mirrors the structure of `TestGroupService_LeaveGroup_LastAdminProtection`.

## Frontend

- New `isLastAdminMember(member)` predicate on `GroupSettingsView.vue`, scoped by the *target's* role plus the group's admin count — **not** by the viewer — so the gate is correct regardless of who is performing the action.
- Remove button renders disabled with `aria-disabled="true"`, a `title` tooltip, and an `aria-describedby`-linked `sr-only` sibling (matches the Leave Group pattern from #1270). The sr-only description is itself gated on `isAdmin && adminCount === 1` so it never sits in the a11y tree when no disabled button references it.
- `data-testid="remove-member-btn-{user_id}"` on each button for deterministic selection from tests.

## Testing

- **Go service test** (`go/services/group_service_test.go`): upgraded & extended as above.
- **Frontend vitest** (`GroupSettingsView.spec.ts`, +4 cases in a new `describe`): disabled on sole admin with tooltip text, enabled on non-admin members, enabled on both admins once two exist, regression guard that clicks on the disabled button don't invoke `removeMember`.
- **Playwright e2e** (`e2e/tests/groups.spec.ts`, +2 tests):
  - API-level: `DELETE /api/v1/groups/{id}/members/{userId}` on the sole admin returns 422 and membership is preserved (atomic rejection).
  - UI-level: fresh group → settings page → Remove button for the sole admin has `disabled`, `aria-disabled="true"`, and the expected `title`.

## Acceptance criteria

- [x] Backend refuses the operation with descriptive error (pre-existing, now directly covered by `ErrorIs`)
- [x] Frontend disables the Remove action on the last admin
- [x] Error surfaced clearly in UI (tooltip + sr-only description; inline error still rendered for any backend-only edge case via existing `error` ref)
- [x] Covered by e2e tests (API + UI)

## Test plan

- [ ] `cd go && go test ./services/...`
- [ ] `cd frontend && npx vitest run`
- [ ] `cd frontend && npm run build`
- [ ] `cd frontend && npx stylelint src/views/groups/GroupSettingsView.vue`
- [ ] `cd e2e && npx tsc --noEmit`
- [ ] Playwright: `cd e2e && npm run test -- groups.spec.ts` with the stack running

### CODE REVIEW

File: frontend/src/views/groups/__tests__/GroupSettingsView.spec.ts [hunk 1]
```diff
@@ -207,3 +207,84 @@ describe('GroupSettingsView — Leave Group action', () => {
...
+    const wrapper = await mountView([
+      makeMembership('user-1', 'admin'),
+      makeMembership('user-2', 'user'),
+      makeMembership('user-3', 'user'),
+    ])
+
+    const soleAdminBtn = wrapper.get('[data-testid="remove-member-btn-user-1"]')
+    expect(soleAdminBtn.attributes('disabled')).toBeDefined()
+  })
+
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/views/groups/__tests__/GroupSettingsView.spec.ts]: Addressed in aa9e63d — dropped the test. As you noted, the `v-if="isAdmin"` guard makes the target-not-viewer case unreachable in this component, and the remaining cases already cover the target-side predicate.

File: frontend/src/views/groups/GroupSettingsView.vue [hunk 2]
```diff
@@ -57,10 +57,29 @@
...
+                class="btn btn-danger btn-small"
+                :data-testid="`remove-member-btn-${member.member_user_id}`"
+                @click="removeMember(member.member_user_id)"
+              >
+                Remove
+              </button>
             </div>
           </div>
         </div>
+        <span id="remove-last-admin-desc" class="sr-only">{{ REMOVE_LAST_ADMIN_TOOLTIP }}</span>
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/views/groups/GroupSettingsView.vue]: Addressed in aa9e63d — the `sr-only` span is now gated on `isAdmin && adminCount === 1` so it only renders when a disabled Remove button actually references it.

File: e2e/tests/groups.spec.ts [hunk 3]
```diff
@@ -397,6 +397,166 @@ test.describe('Main Currency dropdown (#1256)', () => {
...
+
+    try {
+      const membersResp = await request.get(`/api/v1/groups/${groupId}/members`, {
+        headers: {
+          'Accept': 'application/vnd.api+json',
+          'Authorization': `Bearer ${authToken}`,
+        },
+      });
+      const membersBody = await membersResp.json();
+      const admin = membersBody.data.find((m: { attributes: { role: string } }) => m.attributes.role === 'admin');
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply e2e/tests/groups.spec.ts:529]: Addressed in aa9e63d — added `expect(admin).toBeDefined()` before dereferencing, matching the API test.
